python/vowpalwabbit/sklearn_vw.py

```diff
@@ -672,6 +673,73 @@ class VWRegressor(VW, RegressorMixin):
 
     pass
 
+class ThresholdingMultiClassifierMixin(ClassifierMixin):
+
+
+    def predict(self, X):
+        """Predict class labels for samples in X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Samples.
+
+        Returns
+        -------
+        C : array, shape = [n_samples]
+            Predicted class label per sample.
+        """
+        scores = self.decision_function(X)
+        indices = scores.argmax(axis=1)+1
+        return indices
+
+
+class VWMultiClassifier(SparseCoefMixin, ThresholdingMultiClassifierMixin, VW):
+    """Vowpal Wabbit MultiClassifier model """
+
+    def __init__(self, **params):
+
+        params['probabilities'] = True
+        super(VWMultiClassifier, self).__init__(**params)
+
+    def predict(self, X):
+        """Predict class labels for samples in X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Samples.
+
+        Returns
+        -------
+        C : array, shape = [n_samples]
+            Predicted class label per sample.
+        """
+
+        return ThresholdingMultiClassifierMixin.predict(self, X=X)
+
+    def decision_function(self, X):
```

Comment:
it would be more accurate to use predict_proba instead of decision_function.  my understanding is predict_proba provides probabilities for each class and decision_function provides the distance to the decision boundary. So since vw is providing probabilities that should be the right one here.

Reply:
okay